extension-requests/script.js

```diff
@@ -0,0 +1,221 @@
+    { title: 'Created At', key: 'timestamp', time: true },
+    { title: 'Task', key: 'taskId' },
+  ];
+
+  const updateRequestBtn = createElement({
+    type: 'button',
+    attributes: { class: 'update_request' },
+    innerText: 'Update Request',
+  });
+  const moreInfoBtn = createElement({
+    type: 'button',
+    attributes: { class: 'more' },
+    innerText: 'More',
+  });
+  updateRequestBtn.addEventListener('click', () => {
+    showModal('update-form');
+    state.currentExtensionRequest = data;
+    fillUpdateForm();
+  });
+  moreInfoBtn.addEventListener('click', () => {
+    showModal('info');
+    showTaskDetails(data.taskId, data.status === 'APPROVED');
+    state.currentExtensionRequest = data;
+  });
+
+  const main = createTable(dataHeadings, data, 'extension-request');
+
+  main.appendChild(moreInfoBtn);
+  main.appendChild(updateRequestBtn);
+  return main;
+}
+render();
+
+//PATCH requests functions
+async function onStatusFormSubmit(e) {
+  e.preventDefault();
+  try {
+    addLoader(container);
+    let formData = formDataToObject(new FormData(e.target));
+    await updateExtensionRequestStatus({
+      id: state.currentExtensionRequest.id,
+      body: formData,
+    });
+    reload();
+  } catch (error) {
+    errorHeading.textContent = 'Something went wrong';
+    errorHeading.classList.add('error-visible');
+    reload();
+  } finally {
+    removeLoader();
+  }
+}
+async function onUpdateFormSubmit(e) {
+  e.preventDefault();
+  try {
+    addLoader(container);
+    let formData = formDataToObject(new FormData(e.target));
+    formData['newEndsOn'] = new Date(formData['newEndsOn']).getTime() / 1000;
+    await updateExtensionRequest({
+      id: state.currentExtensionRequest.id,
+      body: formData,
+    });
+    reload();
+  } catch (error) {
+    errorHeading.textContent = 'Something went wrong';
+    errorHeading.classList.add('error-visible');
+    reload();
+  } finally {
+    removeLoader();
+  }
+}
+
+modalUpdateForm.addEventListener('submit', onUpdateFormSubmit);
+modalStatusForm.addEventListener('submit', onStatusFormSubmit);
+
+modalParent.addEventListener('click', hideModal);
+closeModal.forEach((node) => node.addEventListener('click', () => hideModal()));
+
+function showModal(show = 'form') {
+  modalParent.classList.add('visible');
+  modalParent.setAttribute('show', show);
+}
+function hideModal(e) {
+  if (!e) {
+    modalParent.classList.remove('visible');
+    return;
+  }
+  e.stopPropagation();
+  if (e.target === modalParent) {
+    modalParent.classList.remove('visible');
+  }
+}
+function reload() {
+  setTimeout(() => window.history.go(0), 2000);
+}
+function fillStatusForm() {
+  modalStatusForm.querySelector('.extensionId').value =
+    state.currentExtensionRequest.id;
+  modalStatusForm.querySelector('.extensionTitle').value =
+    state.currentExtensionRequest.title;
+  modalStatusForm.querySelector('.extensionAssignee').value =
+    state.currentExtensionRequest.assignee;
```

Comment:
we can de structure values from `state.currentExtensionRequest` in the beginning and then use the values in all these lines.